### DISCUSSION TITLE
PrimeVue DataTable paginator page number

### DISCUSSION THREAD

**Discussion by rtarassov** [primefaces/community#395] [PrimeVue] [answered]
Hello

I wanted to make the DataTable remember which page I was on and which rows were displayed and how many rows-per-page i had selected to display.
Everything is fine except the first page number is always being highlighted when I return to the page that has the DataTable.
For example I was on page 2 and the page number 2 was highlighted, had 20 rows-per-page selected (default is 10) and the paginator was displaying page 2. When I go on another page and then return to the page that has the DataTable, it perfectly still displays the page 2 with the correct 20 rows, but in the paginator it highlights page number 1, but I am on the 2nd page and 2nd page rows are displayed.
All the requests-responses include the correct pageNumber etc.

Has anybody dealt with this kind of issue before? What other information should I provide?
```
<DataTable
    id="dataTable-id"
    class="p-datatable-sm"
    v-bind:value="page?.content || []"
    v-bind:paginator="true"
    v-bind:rows="params.size"
    v-bind:rows-per-page-options="[10, 20, 30]"
    v-bind:lazy="true"
    v-bind:loading="loading"
    v-bind:showLoader="true"
    v-bind:total-records="page?.totalElements"
    v-bind:responsive-layout="'scroll'"
    v-bind:sort-field="getSortingPreference().field"
    v-bind:sort-order="getSortingPreference().order"
    v-bind:row-hover="true"
    v-on:page="load($event)"
    v-on:sort="sort($event)"
  >
```

**Comment by tugcekucukoglu** [comment, marked as answer]
`first` property can be controlled via page and sort emits. Here is an example: https://stackblitz.com/edit/sz2c5q?file=src%2FApp.vue%3AL7

**Reply by rtarassov** [reply]
Thank you very much this was exactly what I was missing!
